AppBar 부분 검색 UI 및 라우팅

KimJye opened the pull request:
모바일에서, 검색 아이콘을 눌렀을 때 검색이 활성화됩니다.
모바일, 데스크탑 둘다 검색을 했을떄,  검색 결과 화면으로 라우팅됩니다.
아직 검색 결과 화면은 구현하지 않아서, AppBar부분의 검색 UI와 라우팅 부분만 봐주시면 됩니다.
감사합니다

KimJye committed:
[client] | feat(search): 모바일 검색 활성화 UI 뼈대 작업

모바일 검색 활성화시 요소들에대한 화면 구성을 했습니다.
ArrawBack은 뒤로가기 버튼입니다.
InputLayer는 검색창입니다.

KimJye committed:
[client] | feat(ArrawBackSVG): 뒤로가기 SVG 추가

모바일 검색 활성화시 뒤로가기 작업의 SVG를 추가하였습니다.

KimJye committed:
[client] | feat(search): 모바일 검색 활성화 UI

디자인을 보면서, 모바일 검색 활성화의 UI를 구현했습니다.

KimJye committed:
[client] | feat(search): media로 뒤로가기버튼 display 조절

데스크톱일때는 뒤로가기버튼이 보이면 안되기때문에, 데스크톱일떄는 display none 처리를하였습니다.

KimJye committed:
[client] | feat(search): 검색 결과에 따른 주제지정

검색결과 제목 변경했습니다.

KimJye committed:
merge conflict

KimJye committed:
[client] | feat(AppBar): 모바일 검색UI 구성

모바일화면에서 검색 아이콘을 눌렀을 때, 검색 UI가 활성화됩니다.
또한, 뒤로가기 아이콘을 눌렀을 떄, 기존의 모바일 AppBar상태로
돌아갑니다.

KimJye committed:
[client] | feat(search): 모바일과 데스크탑의  검색바 UI 구현

AppBar에서 하나의 Search 컴포넌트와 상태관리를 통해 구현했습니다.

KimJye committed:
[client] | refactor(search): input width 수정

KimJye committed:
[client] | feat(search): 검색UI에 입력 후 search page 라우팅

onKeyPress의 이벤트로 검색UI에 입력 후 Enter하면useRouter를 이용하ㅕ
검색된 값을 라우팅합니다.

KimJye committed:
[client] | fix(AppBar): 모바일 검색활성화후 데스크탑 로고 없는 버그수정

모바일 검색 활성화 후, 뒤로가기 아이콘을 누르고 데스크탑으로 돌아가면
로고가 있지만, 뒤로가기 아이콘을 누르지않은 상태에서 데스크탑으로
돌아갈때 로고가 사라지고 검색UI가 맨 앞에 있는 버그가 있었습니다.

이 문제를 해결하기 위해서 media를 이용하여 데스크탑일때 Logo가
보여지도록 display block처리를 하였습니다.

KimJye commented on packages/client/views/Search/index.tsx:
```diff
@@ -0,0 +1,44 @@
+        <S.Title>
+          <SearchSVG />
+          <span>"{router.query.query}" 검색 결과</span>
+        </S.Title>
```
이하 아직 미구현된 부분입니다. 앞으로는 리뷰가 필요한 부분만 올리겠습니다.

hyokualexkwon replied:
아직 스케폴딩이니까 올리셔도 괜찮을것 같아요~

KimJye commented on packages/client/views/Search/styles.tsx:
```diff
@@ -0,0 +1,81 @@
+import styled from 'styled-components';
```
아직 미구현된 부분입니다. 앞으로는 리뷰가 필요한 부분만 올리겠습니다.

Atercatus commented on packages/client/components/Search/index.tsx:
```diff
@@ -0,0 +1,37 @@
+import * as S from './styles';
+
+interface Searchprops {
+  inActive: () => void;
```
함수의 네이밍은 동사가 되면 좋을거 같습니다.

KimJye replied:
넵

Atercatus commented on packages/client/components/Search/index.tsx:
```diff
@@ -0,0 +1,37 @@
+  isActive: boolean;
+}
+
+export const Search: React.FunctionComponent<Searchprops> = props => {
```
매개변수를 인터페이스로 정의한김에 destructing을 사용해서 `props`를 중복해서 쓰지 않도록 하면 더 깔끔해질거 같습니다.

KimJye committed:
[client] | refactor(search): pr 리뷰 반영

비활성화하는 함수의 이름을 동사형으로 변경하였습니다.
props를 중복해서 쓰지않기 위해destructing을 사용하였습니다.

Atercatus commented on packages/client/components/AppBar/index.tsx:
```diff
@@ -48,7 +48,7 @@ export const AppBar: React.FunctionComponent<AppBarProps> = ({ maxWidth }) => {
         </S.MobileButtons>
 
-        <Search inActive={handleActiveSearchBar} isActive={isActiveSearchBar} />
+        <Search deActive={handleActiveSearchBar} isActive={isActiveSearchBar} />
```
`deActive` 가 하나의 동사라면 `deactivate` 가 맞는거 같아요

KimJye committed:
[client] | refactor(search): 함수이름 변경

KimJye committed:
[client] | refactor(search): 기존에 작성된 routePath로 변경

Router의 pathname을 string으로 작성했었는데, 기존에 작성된 routePath의
search로 변경했습니다.

Atercatus commented on packages/client/components/AppBar/index.tsx:
```diff
@@ -48,7 +48,7 @@ export const AppBar: React.FunctionComponent<AppBarProps> = ({ maxWidth }) => {
         </S.MobileButtons>
 
-        <Search deActive={handleActiveSearchBar} isActive={isActiveSearchBar} />
+        <Search deactive={handleActiveSearchBar} isActive={isActiveSearchBar} />
```
`deactive ` 란 말은 없고  `deactivate` 가 옳은 표현 이라고 하네요 ㅠㅠ

KimJye committed:
[client] | refactor(search): 함수명 변경

hyokualexkwon commented on packages/client/svgs/SearchSVG/index.tsx:
```diff
@@ -1,7 +1,7 @@
 
 const SvgComponent = props => (
-  <svg width={23} height={24} {...props}>
+  <svg width={20} height={20} {...props}>
```
SearchSVG의 크기가 고정되어서 재사용이 어려워졌네요.
![image](https://user-images.githubusercontent.com/17868599/69200666-28b62800-0b7f-11ea-9816-06a7cc52bb2c.png)
 본 UI에서는 SearchSVG가 가로세로 24px이어야 합니다.

hyokualexkwon commented on packages/client/components/Search/index.tsx:
```diff
@@ -0,0 +1,41 @@
+
+      <S.InputLayer isActive={isActive}>
+        <SearchSVG />
+        <S.Input placeholder="검색" onKeyPress={sendQuery}></S.Input>
```
![image](https://user-images.githubusercontent.com/17868599/69200852-cc073d00-0b7f-11ea-82f6-0f38b0d52221.png)
 input의 영역이 전체 넓이를 포함하고 있지 않습니다.

riverandeye replied:
모바일 환경에서도 입력한 글자가 끝까지 나타나지 않는 오류가 생깁니다.

hyokualexkwon commented on packages/client/components/Search/styles.tsx:
```diff
@@ -0,0 +1,45 @@
+export const Input = styled.input`
+  background-color: #484c50;
+  vertical-align: middle;
+  font-size: 1.8rem;
```
디자인 스펙 상 16px입니다.

hyokualexkwon commented on packages/client/components/Search/styles.tsx:
```diff
@@ -0,0 +1,45 @@
+  height: 3.2rem;
+  background-color: #484c50;
+  border-radius: 0.5rem;
+  padding-top: 0.2rem;
```
이 값을 왜 주신거에요?

hyokualexkwon commented on packages/client/components/Search/styles.tsx:
```diff
@@ -0,0 +1,45 @@
+  border-radius: 0.5rem;
+  padding-top: 0.2rem;
+  padding-right: 1rem;
+  padding-bottom: 0.2rem;
```
이 값을 왜 주신거에요?

hyokualexkwon commented on packages/client/components/Search/index.tsx:
```diff
@@ -0,0 +1,41 @@
+        <ArrowBackSVG />
+      </S.ArrowBack>
+
+      <S.InputLayer isActive={isActive}>
```
마크업 구성을
```
<S.Input>
  <SearchSVG />
  <input />
</S.Input>
```

이런식으로 가져가는건 어떨까요? input엘리먼트에 대해 별도 props로 조작하는 부분이 없으니 따로 styled-component로 빼지 않고 간결하게 작성 할 수 있을것 같아요.

hyokualexkwon commented on packages/client/components/Search/index.tsx:
```diff
@@ -0,0 +1,41 @@
+  isActive,
+}) => {
+  const sendQuery = e => {
+    if (e.key === 'Enter') {
```
모바일에서 테스트 했을때에도 잘되나용?

riverandeye replied:
제가 확인해봤는데 입력 버튼 누르면 검색 페이지 전환 잘 됩니다.

KimJye replied:
> 모바일에서 테스트 했을때에도 잘되나용?

넹~

hyokualexkwon commented on packages/client/components/Search/index.tsx:
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import Router from 'next/router';
```
저희 Router대신에 useRouter를 사용하는건 어떨까요? 저희 앱에서 hook을 사용할 수 있으면 최대한 사용하는게 좋을것 같아서요.

hyokualexkwon commented on packages/client/components/AppBar/index.tsx:
```diff
@@ -1,24 +1,32 @@
 }
 
 export const AppBar: React.FunctionComponent<AppBarProps> = ({ maxWidth }) => {
+  const [isActiveSearchBar, setActiveSearchBar] = useState(false);
```
영미권 사람들은 헷갈려할것 같아요. Is the search bar active?  라고 물어볼것 같은데요? isSearchBarActive가 더 낫지 않을까요?

hyokualexkwon commented on packages/client/components/Search/index.tsx:
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
```
Search라는 이름보다는 SearchBar가 조금 더 ui를 잘 드러내는 네이밍일것 같습니다!

riverandeye commented on packages/client/components/Search/styles.tsx:
```diff
@@ -0,0 +1,45 @@
+  background-color: #484c50;
+  vertical-align: middle;
+  font-size: 1.8rem;
+  margin-left: 1.17rem;
```
디자인 스펙상 1rem이 맞는것 같습니다.

riverandeye reviewed (approved):
라우터를 이용해서 핵심 기능을 적절히 구현한 것으로 보여 Approve 하겠습니다.
리뷰한 부분만 한번 살펴주시면 좋을것 같아요.

KimJye committed:
[client] | refactor(SearchBar): PR리뷰 반영

함수명isActiveSearchBar에서 isSearchBarActive로 수정했습니다.
컴포넌트명을 Search에서 SearchBar로 수정했습니다.
Router대신에 useRouter로 수정했습니다.
검색 입력 부분의 마크업 구성을 변경했습니다.
input 영역이 전체 넓이를 포함하도록 수정했습니다.
불필요한 padding값들을 삭제했습니다.
SVG의 값을 동적으로 변할수있도록 수정했습니다.

Atercatus commented on packages/client/components/AppBar/index.tsx:
```diff
@@ -1,24 +1,32 @@
 export const AppBar: React.FunctionComponent<AppBarProps> = ({ maxWidth }) => {
+  const [isSearchBarActive, SearchBarActive] = useState(false);
+
+  const handleActiveSearchBar = () => {
```
이 부분만 수정하시면 될 거같아욥

hyokualexkwon commented on packages/client/components/AppBar/index.tsx:
```diff
@@ -1,24 +1,32 @@
 }
 
 export const AppBar: React.FunctionComponent<AppBarProps> = ({ maxWidth }) => {
+  const [isSearchBarActive, SearchBarActive] = useState(false);
```
`[isSearchBarActive, setSearchBarActive]` 라고 되어야 할것 같은데요?

KimJye replied:
뜨헉

KimJye committed:
[client] | feat(SearchBar): pr 리뷰 코드 수정

KimJye committed:
머지 충돌 해결

Atercatus commented on packages/client/components/SearchBar/index.tsx:
```diff
@@ -0,0 +1,42 @@
+
+import * as S from './styles';
+
+interface SearchBarprops {
```
`SearchBarProps`로 오타가 난 거 같아요.

Atercatus commented on packages/client/components/SearchBar/index.tsx:
```diff
@@ -0,0 +1,42 @@
+      </S.ArrowBack>
+
+      <S.Input isActive={isActive}>
+        <SearchSVG width="20" height="20" viewBox="0 0 20 20" />
```
`<SearchSVG width={23} height={24} />` 위에는 이런식으로 되어있는데 통일이 되면 좋을 거 같아요.

KimJye replied:
> `<SearchSVG width={23} height={24} />` 위에는 이런식으로 되어있는데 통일이 되면 좋을 거 같아요.

viewBox는 어떻게할수있을까요??ㅠㅠ

Atercatus replied:
```
 <SearchSVG width={20} height={20} viewBox={"0 0 20 20"} />
```
이렇게는 안되나여?

KimJye committed:
[client] | refactor(SearchBar): PR 리뷰 오타수정
